### PR TITLE
CAS-7319, added several new methods to MSMetaData

### PR DESCRIPTION
**Pull request by dmehring**
added various features to MSMetaData

	modified:   ms/MSOper/MSKeys.cc
	modified:   ms/MSOper/MSKeys.h
	modified:   ms/MSOper/MSMetaData.cc
	modified:   ms/MSOper/MSMetaData.h
	modified:   ms/MSOper/MSSummary.cc
	modified:   ms/MSOper/test/tMSMetaData.cc

### CODE REVIEW

**Comment by tammojan** [comment]
@dmehring, could you pull the master into your branch? (`git pull origin master`) Due to a little accident the master would not pass the travis test. If you pull the master and then commit it on your branch, the travis tests should restart automatically.

**Comment by dmehring** [comment]
git pull origin master

Done.

**Comment by juliantaylor** [comment]
you can restart travis jobs in their ui, should be the arrow going in a circle in the build job.
travis should then merge against current master.

closing and reopening the pull also works

I wouldn't recommend merging master into PR's to update them, instead I would rebase them onto the origin branch, this has the advantage that the log is not filled with too many criss-cross merges.

**Comment by dmehring** [comment]
Julian,
I'm not clear if your previous message was directed toward me or someone else. If you all need me to do something else, please be specific as both github and git are very new to me.
Thanks

**Comment by juliantaylor** [comment]
restarting the job in the travis ui would have to be done by someone with commit rights. Though you should be able to close and reopen this PR to archive the same effect via the buttons below the comment box

**Comment by dmehring** [comment]
"Though you should be able to close and reopen this PR to archive the same effect via the buttons below the comment box"

Is this something I need to do?

**Comment by tammojan** [comment]
@dmehring Thanks for pulling in the master (perhaps not the best thing but I wouldn't advise a `git rebase` on your first try at git). I think you did not commit this change yet (`git commit`), or forgot to push it to github (`git push origin`).

I'll restart the test like @juliantaylor suggested once your commit shows up.

**Comment by dmehring** [comment]
@tammojan My apologies. This is my first ever use of GitHub and my first use of git in over six years. I only did
git pull origin master

Would you like me to do both
git commit
andgit push origin

Or perhaps just one of these? Obviously, I'm not yet git literate.

**Comment by tammojan** [comment]
No problem. I think just `git push origin` should suffice (I was a bit confused myself about the commit).

**Comment by dmehring** [comment]
git push origin

Done

**Comment by tammojan** [comment]
@juliantaylor Do you have an idea why Dave's merge doesn't show up here? I'm signing off for today.

**Comment by juliantaylor** [comment]
origin in our current (draft) documentation is the users fork, upstream the main casacore repo.
So to update the branch with the latest master should be:

```
# check that the working copy is clean:
git status
# if dirty you could use git stash to temporarily stash away your changes
git fetch upstream
# check that this contains commit bfca97387f74ee717164aaa632e2925f4dd39129
git log upstream/master
# checkout the branch with your changes
git checkout CAS-7319
# rebase your branch onto the latest upstream/master
git rebase upstream/master
# git log should now show your changes on top of bfca97387f74ee717164aaa632e2925f4dd39129
# push the changes to your github fork named origin
git push -f origin CAS-7319
```

concerning git push origin, this is a bit dangerous as it will push all branches to the remote before git 2.0. I recommend to always quality what you want to push via:

    git push <remote> <branch-to-push>

**Comment by tammojan** [comment]
Actually, @juliantaylor is right and a rebase is better. Since the merge is probably not committed this shouldn't be hard to do: revert the uncommitted changes, then rebase, then commit. @dmehring, could you ask your local git expert to look over your shoulder?

**Comment by dmehring** [comment]
@tammojan I'm afraid @juliantaylor is the closest thing I have to a local git expert, and he of course is a full ocean plus half a continent away.

**Comment by juliantaylor** [comment]
if you like we can go through it via e.g. IRC or some other more direct communication, you can find me as jtaylor in freenode and OFTC, though I'll be traveling until next monday.

**Comment by dmehring** [comment]
@juliantaylor yes please. I'll contact you next week. Thanks!

**Comment by gervandiepen** [comment]
I've merged the pull request.

File: ms/MSOper/MSMetaData.cc [hunk 1]
```diff
@@ -2074,17 +2213,22 @@ std::map<uInt, Double> MSMetaData::getAverageIntervalsForScan(
...
-	return scanSpwToIntervalMap[scan];
-
+	std::map<uInt, Double> ret;
+	std::map<std::pair<ScanKey, uInt>, Double>::const_iterator iter = scanSpwToAverageIntervalMap.begin();
+	std::map<std::pair<ScanKey, uInt>, Double>::const_iterator end = scanSpwToAverageIntervalMap.end();
+	while (iter != end) {
+		if (iter->first.first == scan) {
+			ret[iter->first.second] = iter->second;
+		}
+	}
```
Comment (juliantaylor) [inline ms/MSOper/MSMetaData.cc]: this loop will never end, `iter++` missing?

**Pull request merged**